clippy_lints/src/eta_reduction.rs

```diff
@@ -174,9 +174,9 @@ fn match_types(lhs: &ty::TyKind<'_>, rhs: &ty::TyKind<'_>) -> bool {
 
 fn get_type_name(cx: &LateContext<'_, '_>, kind: &ty::TyKind<'_>) -> String {
     match kind {
-        ty::Adt(t, _) => cx.tcx.item_path_str(t.did),
+        ty::Adt(t, _) => cx.tcx.def_path_str(t.did),
         ty::Ref(_, r, _) => get_type_name(cx, &r.sty),
-        _ => kind.to_string(),
+        _ => unimplemented!(), //kind.to_string(),
```

Comment:
We could revive https://github.com/rust-lang/rust/pull/58140/files#diff-abbb67221915e225569386f89859e8dcL1225 but that would be terrible hack.